packages/icon-button/src/IconButton.tsx

```diff
@@ -82,37 +85,43 @@ const iconButtonSizes = {
   `,
 } as const;
 
-const iconButtonMode = {
-  [Mode.Light]: css`
-    &:hover {
-      color: ${uiColors.gray.dark2};
-
-      &:before {
-        background-color: ${uiColors.gray.light2};
+const getIconButtonModeStyle = (
+  isTouchDevice: boolean,
+  mode: 'dark' | 'light',
+) => {
+  if (mode === Mode.Light) {
+    return css`
+      &: ${isTouchDevice ? 'active' : 'hover'} {
```

Comment:
Why not simply do `&:active, &:hover {}` here? Is there a reason we wouldn't want to persist the hover state when it's being pressed on desktop? It could also eliminate the need for this to be a function.

Also, while it _probably_ works, I'd remove the extra space between the colon and interpolated statement.

Reply:
Updated!